astropy/cosmology/io/html.py

```diff
@@ -0,0 +1,161 @@
+        Cosmology does NOT have a keyword-only argument (e.g. ``**kwargs``).
+        Arguments moved to the metadata will be merged with existing metadata,
+        preferring specified metadata in the case of a merge conflict (e.g. for
+        ``Cosmology(meta={'key':10}, key=42)``, the ``Cosmology.meta`` will be
+        ``{'key': 10}``).
+    cosmology : str or |Cosmology| class or None, optional keyword-only
+        The cosmology class (or string name thereof) to use when constructing
+        the cosmology instance. The class also provides default parameter
+        values, filling in any non-mandatory arguments missing in 'table'.
+    **kwargs : Any
+        Passed to :attr:`astropy.table.QTable.read`.
+        ``format`` is set to 'ascii.html', regardless of input.
+
+    Returns
+    -------
+    |Cosmology| subclass instance
+    """
+    kwargs["format"] = "ascii.html"  # force the format
+    # Reading is handled by `QTable`.
+    with u.add_enabled_units(cu):  # (cosmology units not turned on by default)
+        table = QTable.read(filename, **kwargs)
+
+    # Need to map the table's column names to Cosmology inputs.
+    table_column_names = set(table.columns)
+    for cosmo_name, html_name in _FORMAT_TABLE.items():
+        if html_name in table_column_names:
+            table.rename_column(html_name, cosmo_name)
+
+    # Build the cosmology from table, using the private backend.
+    return from_table(table, index=index, move_to_meta=move_to_meta, cosmology=cosmology)
+
+
+def write_html_table(cosmology: Cosmology, file, *, overwrite=False, cls=QTable, **kwargs):
+    """Serialize the |Cosmology| into a HTML table.
+
+    Parameters
+    ----------
+    cosmology : |Cosmology| subclass instance file : path-like or file-like
+        Location to save the serialized cosmology.
+    file : path-like or file-like
+        Where to write the html table.
+
+    overwrite : bool, optional keyword-only
+        Whether to overwrite the file, if it exists.
+    cls : |Table| class, optional keyword-only
+        Astropy |Table| (sub)class to use when writing. Default is |QTable|
+        class.
+    **kwargs : Any
+        Passed to ``cls.write``.
+
+    Raises
+    ------
+    TypeError
+        If the optional keyword-argument 'cls' is not a subclass of |Table|.
+
+    Examples
+    --------
+    A HTML file containing a Cosmology HTML table should have scripts enabling
+    MathJax.
+
+    ::
+        <script
+        src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
+        <script type="text/javascript" id="MathJax-script" async
+            src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml.js">
+        </script>
+    """
+    # Set cosmology_in_meta as false for now since there is no metadata being kept
+    table = to_table(cosmology, cls=cls, cosmology_in_meta=False)
+
+    cosmo_cls = type(cosmology)
+    n: str
+    for n, col in zip(table.colnames, table.itercols()):

```

Comment:
I'd use `name` instead of the short `n` (here and in next loop). Then the `n: str` would become quite superfluous (for humans at least).

Aside: maybe just use `for name, col in table.columns.items()`

Reply:
Great. Replacing the `zip` is much preferred.

Reply:
✅